ctru-rs/src/services/ir_user.rs

```diff
@@ -0,0 +1,491 @@
+        })
+    }
+
+    /// Try to connect to the device with the provided ID.
+    pub fn require_connection(&self, device_id: IrDeviceId) -> crate::Result<()> {
+        self.send_service_request(
+            vec![REQUIRE_CONNECTION_COMMAND_HEADER, device_id.get_id()],
+            2,
+        )?;
+        Ok(())
+    }
+
+    /// Close the current IR connection.
+    pub fn disconnect(&self) -> crate::Result<()> {
+        self.send_service_request(vec![DISCONNECT_COMMAND_HEADER], 2)?;
+        Ok(())
+    }
+
+    /// Get an event handle that activates on connection status changes.
+    pub fn get_connection_status_event(&self) -> crate::Result<Handle> {
+        let response =
+            self.send_service_request(vec![GET_CONNECTION_STATUS_EVENT_COMMAND_HEADER], 4)?;
+        let status_event = response[3] as Handle;
+
+        Ok(status_event)
+    }
+
+    /// Get an event handle that activates when a packet is received.
+    pub fn get_recv_event(&self) -> crate::Result<Handle> {
+        let response = self.send_service_request(vec![GET_RECEIVE_EVENT_COMMAND_HEADER], 4)?;
+        let recv_event = response[3] as Handle;
+
+        Ok(recv_event)
+    }
+
+    /// Circle Pad Pro specific request.
+    ///
+    /// This will send a packet to the CPP requesting it to send back packets
+    /// with the current device input values.
+    pub fn request_input_polling(&self, period_ms: u8) -> crate::Result<()> {
+        let ir_request: [u8; 3] = [1, period_ms, (period_ms + 2) << 2];
+        self.send_service_request(
+            vec![
+                SEND_IR_NOP_COMMAND_HEADER,
+                ir_request.len() as u32,
+                2 + (ir_request.len() << 14) as u32,
+                ir_request.as_ptr() as u32,
+            ],
+            2,
+        )?;
+
+        Ok(())
+    }
+
+    /// Mark the last `packet_count` packets as processed, so their memory in
+    /// the receive buffer can be reused.
+    pub fn release_received_data(&self, packet_count: u32) -> crate::Result<()> {
+        self.send_service_request(vec![RELEASE_RECEIVED_DATA_COMMAND_HEADER, packet_count], 2)?;
+        Ok(())
+    }
+
+    /// This will let you directly read the ir:USER shared memory via a callback.
+    pub fn process_shared_memory(&self, process_fn: impl FnOnce(&[u8])) {
+        let shared_mem_guard = IR_USER_STATE.lock().unwrap();
+        let shared_mem = shared_mem_guard.as_ref().unwrap();
+
+        process_fn(shared_mem.shared_memory);
+    }
+
+    /// Read and parse the ir:USER service status data from shared memory.
+    pub fn get_status_info(&self) -> IrUserStatusInfo {
+        let shared_mem_guard = IR_USER_STATE.lock().unwrap();
+        let shared_mem = shared_mem_guard.as_ref().unwrap().shared_memory;
+
+        IrUserStatusInfo {
+            recv_err_result: i32::from_ne_bytes([
+                shared_mem[0],
+                shared_mem[1],
+                shared_mem[2],
+                shared_mem[3],
+            ]),
```

Comment:
I simplified this part of the code and after that I don't think it's worthwhile to add the dependency (and I'm parsing the data in a special way for the connection status so that would complicate bytemuck integration).